src/main/java/com/synopsys/integration/blackduck/service/BlackDuckResponsesTransformer.java

```diff
@@ -59,12 +64,14 @@ public BlackDuckResponsesTransformer(BlackDuckHttpClient blackDuckHttpClient, Bl
             allResponses.addAll(blackDuckPageResponse.getItems());
 
             totalCount = blackDuckPageResponse.getTotalCount();
-            if (!getAll) {
+            if (maxPageCount <= 1) {
                 return new BlackDuckPageResponse<>(totalCount, allResponses);
             }
 
-            while (allResponses.size() < totalCount && currentOffset < totalCount) {
+            int currentPageCount = 1;
+            while (allResponses.size() < totalCount && currentOffset < totalCount && currentPageCount < maxPageCount) {
                 currentOffset += pagedRequest.getLimit();
+                currentPageCount += 1;
```

Comment:
I believe there is a shorter notation for this.

Reply:
currentPageCount++ is what I am referring to.

Reply:
You're right.